lib/defaults.js

```diff
@@ -13,19 +15,31 @@ function setContentTypeIfUnset(headers, value) {
   }
 }
 
+/*
+ * Returns a different adapter based on the environment the app is running
+ * On browsers uses an XHR adapter
+ * On a node app uses HTTP adapter
+ */
 function getDefaultAdapter() {
   var adapter;
   if (typeof XMLHttpRequest !== 'undefined') {
-    // For browsers use XHR adapter
     adapter = require('./adapters/xhr');
   } else if (typeof process !== 'undefined' && Object.prototype.toString.call(process) === '[object process]') {
-    // For node use HTTP adapter
-    adapter = require('./adapters/http');
+    adapter = getNodeAdapter();
```

Comment:
although the function will improve,eslint suggests that we should use after define